coral/src/app/main.tsx

```diff
@@ -5,15 +5,36 @@ import router from "src/app/router";
 import "@aivenio/design-system/dist/styles.css";
 import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
 import { ReactQueryDevtools } from "@tanstack/react-query-devtools";
+import { getWindowWithMswInstance } from "src/services/api-mocks/window-msw";
 
 const DEV_MODE = import.meta.env.DEV;
-const queryClient = new QueryClient();
 
-ReactDOM.createRoot(document.getElementById("root") as HTMLElement).render(
-  <React.StrictMode>
-    <QueryClientProvider client={queryClient}>
-      <RouterProvider router={router} />
-      {DEV_MODE && <ReactQueryDevtools />}
-    </QueryClientProvider>
-  </React.StrictMode>
+const root = ReactDOM.createRoot(
+  document.getElementById("root") as HTMLElement
 );
+
+const queryClient = new QueryClient();
+
+function prepare(): Promise<void | ServiceWorkerRegistration> {
+  if (DEV_MODE) {
+    return import("src/services/api-mocks/browser").then(({ worker }) => {
+      if ("start" in worker) {
+        const window = getWindowWithMswInstance();
```

Comment:
Instead of having a function to help with types, could we simply extend the `window` type? I think that would be cleaner approach than a function. I am sure we might have more things bound to window in the future, so we could already set up the practise how to extend that.

```
declare global {
  interface Window {
    msw?: THE_TYPE
  }
}
```

Reply:
I thought about that, but then I didn't want to mess with the global window type 🤔  but rather use it explicit where we need it. But there's no real risk behind having it global I guess?

Reply:
I think there is no risk and we still define property as optional so we should do the undefined checks in the code anyway.